concourse/model/resources.py

```diff
@@ -367,3 +385,26 @@ def __str__(self):
             rp=self.repo_path(),
             b=self.branch()
         )
+
+    def validate(self):
+        super().validate()
+        labels = self.component_descriptor_labels()
+        if not isinstance(labels, Iterable):
```

Comment:
just iterate over them / convert into a list, and rely on the typechecking that is already done there

you may still emit a specialised error, if you see this as a value-add (I do not tbh)

also, I do not think it is exactly appropriate to raise a MVE. ValueError seems to be more appropriate to me